src/search/pagefind.jl

```diff
@@ -0,0 +1,35 @@
+module PageFind
+using NodeJS: NodeJS
+using HypertextLiteral
+
+function inject_script!(custom_scripts, rootpath)
+    pushfirst!(custom_scripts, joinpath("assets", "default", "pagefind_integration.js"))
+    pushfirst!(custom_scripts, joinpath("pagefind", "pagefind.js"))
+end
+
+function inject_styles!(custom_styles)
+    pushfirst!(custom_styles, joinpath("assets", "default", "pagefind.css"))
+end
+
+function render()
+    return @htl """
+    <div class="search nav-item">
+        <input id="search-input" placeholder="Search everywhere...">
+        <ol id="search-result-container" class="suggestions hidden">
+        </ol>
+        <div class="search-keybinding">/</div>
+    </div>
+    """
+end
+
+function build_search_index(root, docs, config, rootpath)
+    if !success(`npx pagefind -V`)
```

Comment:
Should this use `NodeJS_22_jll` or `NodeJS_20_jll`?  Then it's completely self contained...

Reply:
Yes, kinda. 
I'd rather have a pagefind_jll, but building that unfortunately isn't super straightforward.